examples/v1beta1/pipelineruns/alpha/pipelinerun-param-array-indexing.yaml

```diff
@@ -13,21 +13,39 @@ spec:
           value: '$(params.environments[0])'
         - name: environment2
           value: '$(params.environments[1])'
+        - name: environments
+          value: '$(params.environments[*])'
       taskSpec:
         params:
           - name: environment1
             type: string
           - name: environment2
             type: string
+          - name: environments
+            type: array
         steps:
-          # this step should echo "staging"
+          # args must be initialed to "staging"
           - name: echo-params-1
-            image: bash:3.2
+            image: bash:latest
             args: [
-              "echo",
               "$(params.environment1)",
             ]
-          # this step should echo "staging"
+            script: |
+              if [[ $# != 1 ]]; then
+                echo "failed to validate the length of the arguments"
+                echo "Want: 1, Got: $#"
+                exit 1
+              fi
+              if [[ $1 != "staging" ]]; then
+                  echo "failed to validate the one and only argument of the script"
+                  echo "Want: staging, Got: $1"
+                exit 1
+              fi
+              if [[ $(params.environments[2]) != "prod" ]]; then
+                echo "failed to validate indexing into an array param"
+                echo "Want: prod, Got: $(params.environments[2])"
+              fi
```

Comment:
wonder if it makes sense to have this in a separate step? Since this step is mainly used to refer the one param in args and validate it.

Reply:
it could be a step to validate all the elements in the "environments"

Reply:
thank you @Yongxuanzhang 👍 I have updated the example to include an additional step to valid `environments`.